### ISSUE TITLE
Incorrect error.  Function doesn't return anything

### ISSUE THREAD

**Issue by vahpetr** [SonarSource/eslint-plugin-sonarjs#100]
**I want to report a bug**

Remove this use of the output from "flat"; "flat" doesn't return anything.

**Reproducer**

```javascript
function getIds(objects) {
  function* flat(items) {
      for (const item of items) {
        yield* flat(item.children);
        yield item.id;
      }
    }
    return Array.from(flat(objects));
}
```

**Expected behavior**

No error

**Comment by vilchik-elena** [comment]
I can not reproduce with your snippet. This rule should ignore generator functions (see [here](https://github.com/SonarSource/eslint-plugin-sonarjs/blob/078cb7d98f8bbee972f40443834dc19f77fdefb3/src/rules/no-use-of-empty-return-value.ts#L107))
